src/Control.js

```diff
@@ -22,32 +22,66 @@ export class Control extends React.Component {
   state = { instance: null };
 
   mount() {
-    const { type, data, options, state, events, instanceRef } = separateEvents(
-      this.props
-    );
+    const {
+      type,
+      data,
+      options,
+      state,
+      events,
+      items,
+      instanceRef,
+    } = separateEvents(this.props);
     const { ymaps, parent } = this.context;
 
     const instance = new ymaps.control[type]({ data, options, state });
 
     Object.keys(events).forEach(key => addEvent(events[key], key, instance));
     parent.controls.add(instance);
```

Comment:
... and then in this line you can check if `parent` have `add` method (like in [`<GeoObject>` component](https://github.com/gribnoysup/react-yandex-maps/blob/master/src/GeoObject.js#L53)).

If it does, just add it straight to `parent`